pkg/cluster/mpi/service.go

```diff
@@ -9,36 +10,65 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"github.com/dominodatalab/distributed-compute-operator/api/v1alpha1"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/cluster/metadata"
 	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/actions"
 	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/core"
 )
 
 func ServiceWorker() core.OwnedComponent {
-	return &serviceComponent{}
+	return &serviceComponent{
+		comp: ComponentWorker,
+	}
+}
+
+func ServiceClient() core.OwnedComponent {
+	return &serviceComponent{
+		comp: ComponentClient,
+	}
 }
 
-type serviceComponent struct{}
+type serviceComponent struct {
+	comp metadata.Component
+}
 
 func (c serviceComponent) Reconcile(ctx *core.Context) (ctrl.Result, error) {
 	cr := objToMPICluster(ctx.Object)
 
+	ports := []corev1.ServicePort{}
+	var selector map[string]string
+	var extraLabels map[string]string
+	switch c.comp {
+	case ComponentClient:
+		selector = cr.Spec.NetworkPolicy.ClientLabels
+		extraLabels = map[string]string{}
+	case ComponentWorker:
+		ports = append(ports, corev1.ServicePort{
+			Name:       sshdPortName,
+			Port:       sshdPort,
+			TargetPort: intstr.FromString(sshdPortName),
+			Protocol:   corev1.ProtocolTCP,
+		})
+
+		selector = meta.MatchLabelsWithComponent(cr, c.comp)
+		extraLabels = cr.Spec.Worker.Labels
+	case metadata.ComponentNone:
```

Comment:
`default`?

Reply:
I don't really know what default would mean here, because we only create services for specific components so this is undefined and should never happen. I'm actually not sure how to convince Go that this is not needed.

Reply:
I just meant that using `deafult:` here instead of `case metadata.ComponentNone:` would make more sense.